src/engine/MaxConstraint.cpp

```diff
@@ -55,12 +57,29 @@ MaxConstraint::MaxConstraint( const String &serializedMax )
     ++valuesIter;
 
     Set<unsigned> elements;
-    for ( ; valuesIter != values.end(); ++valuesIter )
+    for ( ; strcmp(valuesIter->ascii(), "e"); ++valuesIter )
+    {
         elements.insert( atoi( valuesIter->ascii() ) );
+    }
+
+    // save flag and values indicating eliminated variables
+    ++valuesIter;
+    bool eliminatedVariableFromString =  atoi( valuesIter->ascii() );
+
+    ++valuesIter;
+    double maxValueOfEliminatedFromString = FloatUtils::negativeInfinity();
+
+    if ( eliminatedVariableFromString )
```

Comment:
If I understood correctly, if var==0 then this indicates there are no eliminated variables? But var==0 is a legal value... If you use my comment below for the case where there are no eliminated variables, it will also simplify this.